lib/chat_api/users.ex

```diff
@@ -109,19 +114,36 @@ defmodule ChatApi.Users do
   @doc """
   Gets a single user_profile.
 
-  Raises `Ecto.NoResultsError` if the User profile does not exist.
+  creates new UserProfile if the User profile does not exist.
 
   ## Examples
 
       iex> get_user_profile(123)
       %UserProfile{}
 
-      iex> get_user_profile(456)
-      ** (Ecto.NoResultsError)
-
   """
   def get_user_profile(user_id) do
-    UserProfile |> where(user_id: ^user_id) |> Repo.one() |> Repo.preload(:user)
+    case UserProfile |> where(user_id: ^user_id) |> Repo.one() do
```

Comment:
nitpick -- this might just be my personal preference, but can we instead just pipe directly into the `case` statement? e.g.
```elixir
UserProfile
|> where(user_id: ^user_id)
|> Repo.one()
|> case do
  %UserProfile{} = profile ->
    profile
    |> Repo.preload(:user)

  nil ->
    create_user_profile(user_id)
end
```
I find it a tiny bit easier to read but might just be me :P